### PR TITLE
Yet more eslint rules for CI

### PR DESCRIPTION
**Pull request by reosarevok**
See commits for details

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: script/.check_eslint_rule_config.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    onlyFilesWithFlowAnnotation: true
+  react:
+    version: detect
+    flowVersion: 0.144.0
+
+rules:
+  flowtype/define-flow-type: warn
+  flowtype/use-flow-type: warn
+  react/jsx-uses-vars: warn
+
```
Comment (mwiencek) [inline script/.check_eslint_rule_config.yaml:73]: This shouldn't cause any real problems, but just for a sake of tidyness, there's apparently some trailing whitespace but no newline at the end of the file.

**Review by mwiencek — APPROVED** [review]

File: .eslintrc.yaml [hunk 2]
```diff
@@ -256,6 +256,11 @@ rules:
   react/require-optimization: off
   react/require-render-return: error
   react/self-closing-comp: error
+  react/sort-comp: [warn, {order: [type-annotations,
+                                   static-methods,
+                                   lifecycle,
+                                   everything-else,
+                                   render]}]
```
Comment (yvanzo) [inline .eslintrc.yaml:272]: Instead of changing order for all files but `TagEditor.js`, would it be possible to change order for `TagEditor.js` only?

**Review by yvanzo — COMMENTED** [review]
Just reviewed first commit.

File: .eslintrc.yaml (hunk 2 above)
Comment (reosarevok) [reply .eslintrc.yaml:272]: I guess it depends on whether we have a reason to want this order or not. I didn't pick the order for the rule - @mwiencek, comments? :)

Comment (mwiencek) [reply .eslintrc.yaml:272]: Well, I do think the suggested order is the most natural, but I'm also okay with dropping this rule entirely: class-based components should eventually be converted to functional ones, anyway, and the order doesn't impact readability enough to enforce it IMO.

Comment (reosarevok) [reply .eslintrc.yaml:272]: @yvanzo: enforce the rule for now, or drop the rule for now?

Comment (mwiencek) [reply .eslintrc.yaml]: I dropped it.

**Pull request merged**